Orckestra.Widget.FilteredSelector/Orckestra.Widget.FilteredSelector/WidgetProvider/Functions/SelectFilteredDataWidgetFunction.cs

```diff
@@ -0,0 +1,153 @@
+            string optionsObjectLabelPropertyName,
+            bool required)
+        {
+            if (parameters.TryGetParameter(typeof(SitemapScope).Name, out SitemapScope sitemapScope) == false)
+            {
+                throw new ArgumentException(typeof(SitemapScope).Name);
+            }
+
+            string
+                keySelTag = "KeySelector",
+                statMethTag = "StaticMethodCall",
+                compFuncName = $"{Constants.serializeFuncNamespace}.{Resources.default_text.SerializeMarkupParamsFuncName}";
+
+            XNamespace
+                uc = Namespaces.BindingFormsStdUiControls10,
+                f = Namespaces.BindingFormsStdFuncLib10,
+                bf = Namespaces.BindingForms10,
+                ft = Namespaces.Function10;
+
+            XElement keySelector = StandardWidgetFunctions.BuildBasicFormsMarkup(uc, keySelTag, "Selected", label, helpDefinition, bindingSourceName);
+            keySelector.Add(
+                new XAttribute("OptionsKeyField", optionsObjectKeyPropertyName),
+                new XAttribute("OptionsLabelField", optionsObjectLabelPropertyName),
+                new XAttribute("Required", required));
+
+            XElement keySelectorOptions = new XElement(uc + $"{keySelTag}.Options");
+            keySelector.Add(keySelectorOptions);
+
+            XElement staticMethodCall = new XElement(f + statMethTag);
+            staticMethodCall.Add(
+                new XAttribute("Type", TypeManager.SerializeType(optionsGeneratingStaticType)),
+                new XAttribute("Method", optionsGeneratingStaticMethodName));
+            keySelectorOptions.Add(staticMethodCall);
+
+            XElement staticMethodCallParams = new XElement(uc + $"{statMethTag}.Parameters");
+            staticMethodCall.Add(staticMethodCallParams);
+
+            XElement functionComposer = new XElement(ft + "function");
+            functionComposer.Add(new XAttribute("name", compFuncName));
+            staticMethodCallParams.Add(functionComposer);
+
+            XElement functionParamType = new XElement(ft + "param");
+            functionParamType.Add(
+                new XAttribute("name", Constants.typeNameParamName),
+                new XAttribute("value", optionsGeneratingStaticMethodParameterValue));
+            functionComposer.Add(functionParamType);
+
+            XElement functionPageId = new XElement(ft + "param");
+            functionPageId.Add(new XAttribute("name", Constants.pageIdParamName));
+            functionPageId.Add(new XElement(bf + "read", new XAttribute("source", "PageId")));
+            functionComposer.Add(functionPageId);
+
+            XElement functionSitescope = new XElement(ft + "param");
+            functionSitescope.Add(
+                new XAttribute("name", Constants.sitemapScopeIdParamName),
+                new XAttribute("value", (int)sitemapScope));
+            functionComposer.Add(functionSitescope);

```

Comment:
You can bundle app attributes (and even elements), into XElement constructor, instead of adding each attribute and element one by one.

This way you don't have to have a named variable for every element, and you can build the XElement with one statement:

```
var keySelectorOptions = new XElement(uc + $"{keySelTag}.Options",
	new XElement(f + statMethTag,
		new XAttribute("Type", TypeManager.SerializeType(optionsGeneratingStaticType)),
		new XAttribute("Method", optionsGeneratingStaticMethodName)),
		new XElement(uc + $"{statMethTag}.Parameters", 
			....));

Reply:
Thanks, I have thought dividing into parts will increase readability but maybe it's better to keep it together. Updated